candle-core/src/pickle.rs

```diff
@@ -757,7 +758,16 @@ impl PthTensors {
             tensor_info.dtype,
             &mut reader,
         )?;
-        Ok(Some(tensor))
+
+        if is_fortran_contiguous {
```

Comment:
I think this would only work for tensors that have a "symmetrical" shape. If the shape using the fortran layout is `[d1, d2, d3]`, we should first reshape the tensor that was read assuming C layout to shape `[d3, d2, d1]` and then permute the dimensions.

Reply:
Good catch. I have updated it now to cover other cases. I also added a unit test to confirm it works for that case.